packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js

```diff
@@ -355,9 +368,44 @@ export default class PackMoveOffersCtrl {
     return null;
   }
 
-  getModemOptionLabel(modemOption) {
-    return this.$translate.instant(`pack_move_modem_${modemOption.name}`, {
-      price: modemOption.price ? modemOption.price.text : '',
+  initModemOptions() {
+    this.offers.forEach((offer) => {
+      Object.values(offer.modemOptions).map((option) => {
+        let optionLabel = '';
+        if (!offer.prices.modemRental.price) {
+          optionLabel = this.$translate.instant(
+            `pack_move_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        } else if (
+          this.MODEM_LIST.includes(option.name) &&
+          offer.prices.modemRental.price.value === option.price.value
+        ) {
+          optionLabel = this.$translate.instant('pack_move_modem_keep', {
+            price: option.price ? option.price.text : '',
+          });
+        } else {
+          optionLabel = this.$translate.instant(
+            `pack_move_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        }
+        set(option, 'label', optionLabel);
+        return option;
+      });
```

Comment:
**nitpick:** `Array.prototype.map` returns a new array. But its return value is not used/captured. Please use appropriate array method.

Reply:
OK I understand your feedback, but what will you suggest for appropriate array method?